pypicloud/views/simple.py

```diff
@@ -31,13 +32,23 @@ def upload(request, content, name=None, version=None):
             return request.forbid()
         try:
             return request.db.upload(content.filename, content.file, name=name,
-                                     version=version)
+                                     version=version, summary=summary)
         except ValueError as e:
             return HTTPBadRequest(*e.args)
     else:
         return HTTPBadRequest("Unknown action '%s'" % action)
 
 
+@xmlrpc_method(endpoint='pypi')
+def search(request, criteria, query_type):
+    """
+    Perform searches from pip. This handles XML RPC requests to the "pypi"
+    endpoint (configured as /pypi/) that specify the method "search".
+
+    """
+    return request.db.search(criteria, query_type)
```

Comment:
It's important that the search only returns packages that the user has read access to. You can see several examples of how to do this [in this file](https://github.com/stevearc/pypicloud/blob/8138668402e58e5ecde25369040711de26baed73/pypicloud/views/simple.py#L50). One particular edge case to pay attention to is that with HTTP, pip will only send up the username/password if the server requests it by returning a 403. This means that in a lot of places we check if the user is logged in, and if not we [request the credentials](https://github.com/stevearc/pypicloud/blob/8138668402e58e5ecde25369040711de26baed73/pypicloud/views/simple.py#L156). I don't know what the behavior is with the RPC endpoint, but it's worth checking.